src/transforms/grok_parser.rs

```diff
@@ -79,10 +79,14 @@ impl Transform for GrokParser {
         let mut event = event.into_log();
         let value = event.get(&self.field).map(|s| s.to_string_lossy());
 
+        let mut drop_field = self.drop_field;
         if let Some(value) = value {
             if let Some(matches) = self.pattern.match_against(&value) {
                 for (name, value) in matches.iter() {
```

Comment:
Could `drop_field` be calculated here as:
```
let drop_field = self.drop_field and !matches.contains_key(&self.field);
```

This would elimiate the `let mut`

Reply:
Good call!